In [1]:
import pandas as pd
pd.__version__

u'0.20.2'

In [2]:
# Chapter 5
props = pd.read_csv('../data/committees.csv')
props.head()

ocd_prop_id  calaccess_prop_id  \
0  ocd-contest/b51dc64d-3562-4913-a190-69f5088c22a6            1376258   
1  ocd-contest/b51dc64d-3562-4913-a190-69f5088c22a6            1376258   
2  ocd-contest/b51dc64d-3562-4913-a190-69f5088c22a6            1376258   
3  ocd-contest/b51dc64d-3562-4913-a190-69f5088c22a6            1376258   
4  ocd-contest/85990193-9d6f-4600-b8e7-bf1317841d82            1362198   

   ccdc_prop_id                                          prop_name  \
0            70  PROPOSITION 051 - SCHOOL BONDS. FUNDING FOR K-...   
1            70  PROPOSITION 051 - SCHOOL BONDS. FUNDING FOR K-...   
2            70  PROPOSITION 051 - SCHOOL BONDS. FUNDING FOR K-...   
3            70  PROPOSITION 051 - SCHOOL BONDS. FUNDING FOR K-...   
4            71  PROPOSITION 052 - STATE FEES ON HOSPITALS. FED...   

   ccdc_committee_id  calaccess_committee_id  \
0                382                 1374469   
1                383                 1220380   
2                384                 1282321   
3                385                 1382843   
4                386                 1362973   

                                      committee_name committee_position  
0  YES ON PROPOSITION 51 - CALIFORNIANS FOR QUALI...            SUPPORT  
1  COMMUNITY COLLEGE FACILITY COALITION ISSUES CO...            SUPPORT  
2  TORLAKSON'S INVEST IN CALIFORNIA A BALLOT MEAS...            SUPPORT  
3  CALIFORNIA TAXPAYERS AND EDUCATORS OPPOSED TO ...             OPPOSE  
4  YES ON PROPOSITION 52 - A COALITION OF CALIFOR...            SUPPORT

In [3]:
contribs = pd.read_csv('../data/contributions.csv')
contribs.head()

calaccess_committee_id                              committee_name  \
0                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   
1                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   
2                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   
3                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   
4                 1386560  ADULT USE CAMPAIGN FOR PROPOSITION 64; THE   

   calaccess_filing_id date_received contributor_lastname  \
0              2083796    2016-09-18              BERGMAN   
1              2083796    2016-09-18                KAHLE   
2              2083796    2016-07-15             MCDEVITT   
3              2083796    2016-08-10      WARREN-SAMARIPA   
4              2083796    2016-07-26      WARREN-SAMARIPA   

  contributor_firstname contributor_city contributor_state contributor_zip  \
0              GRETCHEN    SPRING VALLEY                CA           91978   
1                 MYRNA        SAN DIEGO                CA           92109   
2                   LEO        ESCONDIDO                CA           92025   
3             STEPHANIE        SAN DIEGO                CA           92116   
4             STEPHANIE        SAN DIEGO                CA           92116   

        contributor_employer contributor_occupation  \
0                 A NEW PATH     EXECUTIVE DIRECTOR   
1   NATIONAL SCHOOL DISTRICT                TEACHER   
2              LIFE IONIZERS    SEO/CONTENT MANAGER   
3  STEPHANIE WARREN SAMARIPA           ENTREPRENEUR   
4  STEPHANIE WARREN SAMARIPA           ENTREPRENEUR   

   contributor_is_self_employed  amount  
0                         False    84.0  
1                         False    35.0  
2                         False   198.0  
3                         False   -50.0  
4                          True    50.0

In [4]:
props.prop_name.value_counts()

PROPOSITION 057 - CRIMINAL SENTENCES. JUVENILE CRIMINAL PROCEEDINGS AND SENTENCING. INITIATIVE CONSTITUTIONAL AMENDMENT AND STATUTE.                           13
PROPOSITION 056 - CIGARETTE TAX TO FUND HEALTHCARE, TOBACCO USE PREVENTION, RESEARCH, AND LAW ENFORCEMENT. INITIATIVE CONSTITUTIONAL AMENDMENT AND STATUTE.    12
PROPOSITION 064- MARIJUANA LEGALIZATION. INITIATIVE STATUTE.                                                                                                   11
PROPOSITION 066- DEATH PENALTY. PROCEDURES. INITIATIVE STATUTE.                                                                                                 9
PROPOSITION 055 - TAX EXTENSION TO FUND EDUCATION AND HEALTHCARE. INITIATIVE CONSTITUTIONAL AMENDMENT.                                                          8
PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON SINGLE-USE PLASTIC BAGS.                                                                                         7
PROPOSITION 062- DEATH PENAL

In [5]:
prop_64 = props[props.prop_name == 'PROPOSITION 064- MARIJUANA LEGALIZATION. INITIATIVE STATUTE.']
prop_64

ocd_prop_id  calaccess_prop_id  \
74  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
75  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
76  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
77  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
78  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
79  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
80  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
81  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
82  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
83  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
84  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   

    ccdc_prop_id                                          prop_name  \
74            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
75            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
76            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
77            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
78            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
79            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
80            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
81            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
82            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
83            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
84            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   

    ccdc_committee_id  calaccess_committee_id  \
74                456                 1381808   
75                457                 1382525   
76                458                 1376077   
77                459                 1382568   
78                460                 1371855   
79                461                 1343793   
80                462                 1386560   
81                463                 1387789   
82                464                 1385506   
83                465                 1385745   
84                466                 1390962   

                                       committee_name committee_position  
74  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...            SUPPORT  
75                             NEW APPROACH PAC (MPO)            SUPPORT  
76  CALIFORNIANS FOR SENSIBLE REFORM, SPONSORED BY...            SUPPORT  
77  PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...             OPPOSE  
78  MARIJUANA POLICY PROJECT OF CALIFORNIA, YES ON 64            SUPPORT  
79  CALIFORNIANS FOR RESPONSIBLE MARIJUANA REFORM,...            SUPPORT  
80         ADULT USE CAMPAIGN FOR PROPOSITION 64; THE            SUPPORT  
81  SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...             OPPOSE  
82  DRUG POLICY ACTION - NON PROFIT 501C4, YES ON ...            SUPPORT  
83       FUND FOR POLICY REFORM (NONPROFIT 501(C)(4))            SUPPORT  
84                        MOMS STRONG AGAINST PROP 64             OPPOSE

In [6]:
prop_64.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11 entries, 74 to 84
Data columns (total 8 columns):
ocd_prop_id               11 non-null object
calaccess_prop_id         11 non-null int64
ccdc_prop_id              11 non-null int64
prop_name                 11 non-null object
ccdc_committee_id         11 non-null int64
calaccess_committee_id    11 non-null int64
committee_name            11 non-null object
committee_position        11 non-null object
dtypes: int64(4), object(4)
memory usage: 792.0+ bytes


In [7]:
contribs.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 56379 entries, 0 to 56378
Data columns (total 13 columns):
calaccess_committee_id          56379 non-null int64
committee_name                  56379 non-null object
calaccess_filing_id             56379 non-null int64
date_received                   56379 non-null object
contributor_lastname            56375 non-null object
contributor_firstname           53411 non-null object
contributor_city                56369 non-null object
contributor_state               56363 non-null object
contributor_zip                 56366 non-null object
contributor_employer            48572 non-null object
contributor_occupation          53273 non-null object
contributor_is_self_employed    56379 non-null bool
amount                          56379 non-null float64
dtypes: bool(1), float64(1), int64(2), object(9)
memory usage: 5.2+ MB


In [8]:
# merge the contributions and propostions dataset relevant to prop_64
merged = pd.merge(prop_64, contribs, on='calaccess_committee_id')
merged.head()

ocd_prop_id  calaccess_prop_id  \
0  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
1  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
2  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
3  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
4  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   

   ccdc_prop_id                                          prop_name  \
0            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
1            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
2            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
3            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
4            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   

   ccdc_committee_id  calaccess_committee_id  \
0                456                 1381808   
1                456                 1381808   
2                456                 1381808   
3                456                 1381808   
4                456                 1381808   

                                    committee_name_x committee_position  \
0  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...            SUPPORT   
1  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...            SUPPORT   
2  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...            SUPPORT   
3  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...            SUPPORT   
4  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...            SUPPORT   

                                    committee_name_y  calaccess_filing_id  \
0  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...              2038581   
1  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...              2038581   
2  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...              2038581   
3  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...              2038581   
4  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...              2038581   

  date_received                               contributor_lastname  \
0    2016-02-04                                     BRADEN QUIGLEY   
1    2016-01-05  CALIFORNIANS FOR SENSIBLE REFORM, SPONSORED BY...   
2    2016-01-29  CALIFORNIANS FOR SENSIBLE REFORM, SPONSORED BY...   
3    2016-03-28  CALIFORNIANS FOR SENSIBLE REFORM, SPONSORED BY...   
4    2016-01-05                                 DRUG POLICY ACTION   

  contributor_firstname contributor_city contributor_state contributor_zip  \
0                 LAURA       SACRAMENTO                CA           95825   
1                   NaN           IRVINE                CA           92618   
2                   NaN           IRVINE                CA           92618   
3                   NaN           IRVINE                CA           92618   
4                   NaN         NEW YORK                NY           10001   

  contributor_employer contributor_occupation  contributor_is_self_employed  \
0       LBQ STRATEGIES             CONSULTANT                         False   
1                  NaN                    NaN                         False   
2                  NaN                    NaN                         False   
3                  NaN                    NaN                         False   
4                  NaN                    NaN                         False   

     amount  
0     100.0  
1  250000.0  
2  250000.0  
3  250000.0  
4  250000.0

In [9]:
merged.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 860 entries, 0 to 859
Data columns (total 20 columns):
ocd_prop_id                     860 non-null object
calaccess_prop_id               860 non-null int64
ccdc_prop_id                    860 non-null int64
prop_name                       860 non-null object
ccdc_committee_id               860 non-null int64
calaccess_committee_id          860 non-null int64
committee_name_x                860 non-null object
committee_position              860 non-null object
committee_name_y                860 non-null object
calaccess_filing_id             860 non-null int64
date_received                   860 non-null object
contributor_lastname            860 non-null object
contributor_firstname           750 non-null object
contributor_city                860 non-null object
contributor_state               860 non-null object
contributor_zip                 860 non-null object
contributor_employer            657 non-null object
contributor_occu

In [10]:
# total $ spent on campaigns pro and con
merged.amount.sum()

35177017.640000001

In [11]:
# Which side got more large donations?
merged.committee_position.value_counts()

SUPPORT    762
OPPOSE      98
Name: committee_position, dtype: int64

In [12]:
support = merged[merged.committee_position == 'SUPPORT']
support.amount.sum()

32675806.0

In [13]:
oppose = merged[merged.committee_position == 'OPPOSE']
oppose.amount.sum()

2501211.6400000001

In [14]:
support_vs_oppose = support.amount.sum() / merged.amount.sum()
support_vs_oppose

0.92889642704798669

In [15]:
# what were the ten biggest contributors?
top_10_support = support.sort_values('amount', ascending=False)
top_10_support.head(10)

ocd_prop_id  calaccess_prop_id  \
142  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
79   ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
851  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
846  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
852  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
141  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
850  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
143  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
845  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
254  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   

     ccdc_prop_id                                          prop_name  \
142            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
79             83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
851            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
846            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
852            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
141            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
850            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
143            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
845            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
254            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   

     ccdc_committee_id  calaccess_committee_id  \
142                456                 1381808   
79                 456                 1381808   
851                465                 1385745   
846                464                 1385506   
852                465                 1385745   
141                456                 1381808   
850                465                 1385745   
143                456                 1381808   
845                464                 1385506   
254                457                 1382525   

                                      committee_name_x committee_position  \
142  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...            SUPPORT   
79   YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...            SUPPORT   
851       FUND FOR POLICY REFORM (NONPROFIT 501(C)(4))            SUPPORT   
846  DRUG POLICY ACTION - NON PROFIT 501C4, YES ON ...            SUPPORT   
852       FUND FOR POLICY REFORM (NONPROFIT 501(C)(4))            SUPPORT   
141  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...            SUPPORT   
850       FUND FOR POLICY REFORM (NONPROFIT 501(C)(4))            SUPPORT   
143  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...            SUPPORT   
845  DRUG POLICY ACTION - NON PROFIT 501C4, YES ON ...            SUPPORT   
254                             NEW APPROACH PAC (MPO)            SUPPORT   

                                      committee_name_y  calaccess_filing_id  \
142  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...              2083528   
79   YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...              2083528   
851       FUND FOR POLICY REFORM (NONPROFIT 501(C)(4))              2082414   
846  DRUG POLICY ACTION - NON PROFIT 501C4, YES ON ...              2083608   
852       FUND FOR POLICY REFORM (NONPROFIT 501(C)(4))              2096507   
141  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...              2083528   
850       FUND FOR POLICY REFORM (NONPROFIT 501(C)(4))              2082414   
143  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...              2083528   
845  DRUG POLICY ACTION - NON PROFIT 501C4, YES ON ...              2083608   
254                             NEW APPROACH PAC (MPO)              2083758   

    date_received                               contributor_lastname  \
142    2016-09-15                     PARKER AND AFFILIATED ENTITIES   

In [16]:
top_10_oppose = oppose.sort_values('amount', ascending=False)
top_10_oppose.head(10)

ocd_prop_id  calaccess_prop_id  \
839  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
312  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
310  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
308  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
840  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
837  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
838  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
287  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
836  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   
306  ocd-contest/232cb72c-1b11-4293-b4e2-a181e80d172e            1381868   

     ccdc_prop_id                                          prop_name  \
839            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
312            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
310            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
308            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
840            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
837            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
838            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
287            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
836            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   
306            83  PROPOSITION 064- MARIJUANA LEGALIZATION. INITI...   

     ccdc_committee_id  calaccess_committee_id  \
839                463                 1387789   
312                459                 1382568   
310                459                 1382568   
308                459                 1382568   
840                463                 1387789   
837                463                 1387789   
838                463                 1387789   
287                459                 1382568   
836                463                 1387789   
306                459                 1382568   

                                      committee_name_x committee_position  \
839  SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...             OPPOSE   
312  PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...             OPPOSE   
310  PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...             OPPOSE   
308  PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...             OPPOSE   
840  SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...             OPPOSE   
837  SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...             OPPOSE   
838  SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...             OPPOSE   
287  PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...             OPPOSE   
836  SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...             OPPOSE   
306  PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...             OPPOSE   

                                      committee_name_y  calaccess_filing_id  \
839  SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...              2083142   
312  PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...              2098048   
310  PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...              2083071   
308  PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...              2083071   
840  SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...              2083142   
837  SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...              2064155   
838  SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...              2064155   
287  PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...              2064127   
836  SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...              2064155   
306  PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...              2083071   

    date_received                               contributor_lastname  \
839    2016-07-20                                            SCHAUER   

In [17]:
# group the two committee_name_ columns as they are just duplicates due to previous merge
merged.groupby('committee_name_x').amount.sum().reset_index().sort_values('amount', ascending=False)

committee_name_x       amount
10  YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...  15734047.00
4        FUND FOR POLICY REFORM (NONPROFIT 501(C)(4))   6140000.00
3   DRUG POLICY ACTION - NON PROFIT 501C4, YES ON ...   4470000.00
7                              NEW APPROACH PAC (MPO)   3865000.00
9   SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...   1364000.00
8   PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...   1130930.00
1   CALIFORNIANS FOR RESPONSIBLE MARIJUANA REFORM,...    860470.00
2   CALIFORNIANS FOR SENSIBLE REFORM, SPONSORED BY...    850000.00
5   MARIJUANA POLICY PROJECT OF CALIFORNIA, YES ON 64    739200.00
0          ADULT USE CAMPAIGN FOR PROPOSITION 64; THE     17089.00
6                         MOMS STRONG AGAINST PROP 64      6281.64

In [18]:
# Finding the top contributors
merged.groupby(["contributor_firstname", "contributor_lastname"]).amount.sum().reset_index().sort_values("amount", ascending=False)

contributor_firstname            contributor_lastname     amount
483                  SEAN  PARKER AND AFFILIATED ENTITIES  7250000.0
482                  SEAN                          PARKER  1500000.0
281                 JULIE                         SCHAUER  1364000.0
108                DANIEL                           LEWIS  1250000.0
194                 HENRY                   VAN AMERINGEN  1000000.0
405              NICHOLAS                        PRITZKER   900000.0
421                 PETER                           THIEL   300000.0
456                 ROGER                         JENKINS    75000.0
23                 ANGELA                          HOWARD    65000.0
322              LAWRENCE                            HESS    30000.0
221                 JASON                            POST    10000.0
460              ROGER B.                         MCNAMEE    10000.0
548           ZACHARY PYM                        WILLIAMS    10000.0
239                JEREMY                         ZACHARY     7500.0
440                ROBERT                           ADAMS     6500.0
320                LAURIE                        BENENSON     5000.0
63                CAMILLA                           FIELD     5000.0
375               MICHAEL                            DEAN     5000.0
361            MARY QUINN                         DELANEY     5000.0
488             SHARON K.                        CHRISTIE     5000.0
515                 TERRY                          TURNER     3664.0
301              KHURSHID                           KHOJA     3300.0
277                  JULI                        CROCKETT     3000.0
149                  DOUG                        LANPHERE     2920.0
76                CHARLES                           ROVEN     2500.0
158                 EMILY                          PAXHIA     2500.0
414               PATRICK                       MCMANAMON     2500.0
507                STEVEN                          PERSKY     2500.0
382               MICHAEL                       STEINMETZ     2500.0
377               MICHAEL                           KLEIN     2500.0
..                    ...                             ...        ...
75                CHARLES                            LOVE      100.0
191              GRETCHEN                         BERGMAN       84.0
387              MICHELLE                           WOLFF       84.0
385              MICHELLE                          SEXTON       84.0
189                  GREG                           SMITH       75.0
389                  MIKE                         BARNETT       50.0
534                 WENDY                          COLTON       50.0
168                 ETHAN                         BROOKES       50.0
85           CHRISTINE A.                          PORTER       50.0
459                 ROGER                           TOWER       50.0
492                  SKIP                          ROACHE       50.0
335              LJUBOMIR                       BUTUROVIC       50.0
418                 PETER                          KIRSCH       50.0
326               LEONARD                        KAUFMANN       50.0
104                  DANA                          HIRSCH       50.0
105                DANIEL                          DODSON       50.0
125                 DAVID                          MURRAY       50.0
272                  JOSH                       KLEYMEYER       42.0
51                  BRIAN                        MITCHELL       35.0
533                 WAYNE                         SCHERER       30.0
448                ROBERT                         LOCKART       25.0
185                GEORGE                          OKEEFE       25.0
510               SUE ANN                           LORIG       25.0
355                  MARK                        SLABAUGH       25.0
540               WILLIAM                        SANTSCHI       25.0
430                  RICH                        GOLDBERG       25.0
35

## Extra questions to answer:

1. What percentage of donations came from people who live outside of California?
2. What are the top employers of donors who gave for and against the measure?
3. Which committees had the fewest donors?

In [19]:
# Q1 - What percentage of donations came from people who live outside of California?
outside = merged[merged.contributor_state != 'CA']
inside = merged[merged.contributor_state == 'CA']

total = merged.amount.sum()
o = outside.amount.sum()
i = inside.amount.sum()

pct = total / 100
[o / pct, i / pct]

[50.276646022115706, 49.723353977884294]

__Answer #1__

Donations came more or less half and half from Californian (49.72%) and non Californian donations (50.27%)

In [20]:
# Q2 - What are the top employers of donors who gave for and against the measure?

oppose_employers = oppose.contributor_employer.value_counts()
support_employers = support.contributor_employer.value_counts()

[oppose_employers.head(), support_employers.head()]

[BOSTON SCIENTIFIC      5
 RETIRED                4
 STATE OF CALIFORNIA    4
 SELF                   3
 CITY OF SUNNYVALE      2
 Name: contributor_employer, dtype: int64,
 NONE                                       92
 LOS ANGELES DEPARTMENT OF WATER & POWER     7
 RETIRED                                     6
 APPLE, INC.                                 6
 SOUTHERN CALIFORNIA EDISON                  5
 Name: contributor_employer, dtype: int64]

__ Answer #2__

The top five employers for those __opposing__ _Proposition 64_ are:
- BOSTON SCIENTIFIC      5
- RETIRED                4
- STATE OF CALIFORNIA    4
- SELF                   3
- CITY OF SUNNYVALE      2

The top five employers for those __supporting__ _Proposition 64_ are:

- NONE                                       92
- LOS ANGELES DEPARTMENT OF WATER & POWER     7
- RETIRED                                     6
- APPLE, INC.                                 6
- SOUTHERN CALIFORNIA EDISON                  5

In [21]:
# Q3 - Which committees had the fewest donors?

merged.committee_name_x.value_counts().reset_index()

index  committee_name_x
0   CALIFORNIANS FOR RESPONSIBLE MARIJUANA REFORM,...               420
1   YES ON 64, CALIFORNIANS TO CONTROL, REGULATE A...               250
2   PUBLIC AND MENTAL HEALTH ADVOCATES AGAINST 64,...                88
3   MARIJUANA POLICY PROJECT OF CALIFORNIA, YES ON 64                40
4          ADULT USE CAMPAIGN FOR PROPOSITION 64; THE                26
5                              NEW APPROACH PAC (MPO)                 9
6        FUND FOR POLICY REFORM (NONPROFIT 501(C)(4))                 8
7   DRUG POLICY ACTION - NON PROFIT 501C4, YES ON ...                 7
8   SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT...                 6
9                         MOMS STRONG AGAINST PROP 64                 4
10  CALIFORNIANS FOR SENSIBLE REFORM, SPONSORED BY...                 2

__Answer #3__

The following committees had the fewest donnors for _Proposition 64_

- FUND FOR POLICY REFORM (NONPROFIT 501(C)(4))
- DRUG POLICY ACTION - NON PROFIT 501C4, YES ON
- SAM ACTION, INC., A COMMITTEE AGAINST PROPOSIT
- MOMS STRONG AGAINST PROP 64
- CALIFORNIANS FOR SENSIBLE REFORM, SPONSORED BY